### PR TITLE
Feature/sequence anomally detection

### PR DESCRIPTION
**Pull request by gauravakto**
## Summary

Implements sequence-based anomaly detection on top of the API sequence probability model built by mini-runtime.

## Flow

```
mini-runtime (feature/mini-runtime-release)
    → SessionAnalyzer: per-user deque tracks API call sequences
    → WindowAccumulator: counts transitions (A→B) per 10-min tumbling window
    → ApiSequencesFlusher: upserts to MongoDB api_sequences collection
        { paths: ["/login#POST", "/dashboard#GET"], probability: 0.82, ... }

threat-detection (this branch)
    → SequenceCache (new): loads api_sequences from MongoDB every 10 min
        Map<"prevApi|currentApi", probability>

    Per request in MaliciousTrafficDetectorTask.processRecord():
        1. actor = sourceIP (Kafka-partitioned — each IP always hits same instance)
        2. urlForAggregation = templatized URL (e.g. /users/INTEGER)
        3. currentApiKey = "{collectionId} {url} {method}"
        4. checkSequenceAnomaly(actor, currentApiKey)  ← new, in-memory only
              prevApiKey = actorLastApi.get(actor)      ← Caffeine, TTL 10 min
              actorLastApi.put(actor, currentApiKey)
              if prevApiKey == null: skip (first request)
              probability = sequenceCache.get(prevApiKey|currentApiKey)
              if probability == null OR probability < 0.05: anomalous
                  actorAnomalyCount[actor]++
                  if count >= 5:
                      actorAnomalyCount.invalidate(actor)
                      → generateAndPushMaliciousEventRequest(SequenceAnomaly, SINGLE)
                            → internalKafka → KafkaTopic.ThreatDetection.ALERTS
                                → SendMaliciousEventsToBackend → dashboard
```

**Key design decisions:**
- No Redis on the hot path — all state is Caffeine in-memory
- No reset on normal transitions — counter accumulates across the window, only resets after alert or TTL expiry (prevents bypass by mixing normal requests)
- Kafka partitioned by source IP → single instance owns all requests per actor, no cross-instance state sharing needed
- `SequenceCache` owns all detection state (probabilities + per-actor tracking) — independently testable without constructing the full task

## Changes

- **`libs/dao`**: `ApiSequences` DTO/DAO sync with `feature/mini-runtime-release` — removed `@BsonId int id`, added `lastStateCount`, unique compound index on `(apiCollectionId, paths)`
- **`libs/utils`**: `DataActor` / `DbActor` / `DbLayer` / `ClientActor` — added `fetchApiSequences()`
- **`apps/threat-detection`**:
  - `SequenceCache` — transition probability cache + per-actor anomaly tracking
  - `Utils.buildSequenceAnomalyFilter()` — synthetic `FilterConfig` (category=Behavioral, severity=HIGH)
  - `MaliciousTrafficDetectorTask` — wires `SequenceCache`, delegates to `checkSequenceAnomaly()`
  - `SequenceCacheTest` — 11 behavioral tests covering threshold, counter accumulation, actor isolation, boundary conditions

## Related PRs
- database-abstractor `fetchApiSequences` endpoint: akto-api-security/akto#5004 (against `feature/cyborg-release`)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/threat-detection/src/main/java/com/akto/threat/detection/cache/SequenceCache.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        return transitionProbabilities.get(prevApi + "|" + currentApi);
+    }
+
+    public int size() {
+        return transitionProbabilities.size();
+    }
+
+    private void refreshIfStale() {
+        int now = (int) (System.currentTimeMillis() / 1000);
+        if (!transitionProbabilities.isEmpty() && now - lastUpdatedAt < REFRESH_INTERVAL_SEC) {
```
Comment (ayushaga14) [inline apps/threat-detection/src/main/java/com/akto/threat/detection/cache/SequenceCache.java]: initially transitionProbabilities would be empty, this will return false

**Review by notshivansh — DISMISSED** [review]

**Review by notshivansh — APPROVED** [review]
